src/drop.js

```diff
@@ -22,5 +22,5 @@ var _xdrop = require('./internal/_xdrop');
  *      R.drop(3, [1,2,3,4,5,6,7]); //=> [4,5,6,7]
  */
 module.exports = _curry2(_dispatchable('drop', _xdrop, function drop(n, list) {
-  return n < list.length ? _slice(list, n) : [];
+  return n <= 0 ? list : _slice(list, n);
```

Comment:
Below was the argument I was going to make before I started to reconsider:

> I guess my feeling is that if we're going to clone, then we should go all the way:
> 
> ```javascript
> var list = [{x: 1}, {x: 2}, {x: 3}];
> var a = R.head(list);
> var b = R.head(R.take(1, list));
> R.deepEq(a, b); //=> true
> a == b; //=> false, because we've cloned.
> ```
> 
> Perf is only one minor concern.  The major one has to do with what sort of API we're presenting....

But now I can't get my head around it.  I'm not really sure what sounds correct.  I'll think on it a little more.